packages/kbn-alerts-ui-shared/src/common/apis/fetch_alert_fields.ts

```diff
@@ -6,23 +6,27 @@
  * Side Public License, v 1.
  */
 
+import type { FieldDescriptor } from '@kbn/data-views-plugin/server';
+import { HttpSetup } from '@kbn/core-http-browser';
 import { ValidFeatureId } from '@kbn/rule-data-utils';
-import { HttpSetup } from '@kbn/core/public';
-import { FieldSpec } from '@kbn/data-views-plugin/common';
+import type { BrowserFields } from '../types/alerts_fields_types';
 import { BASE_RAC_ALERTS_API_PATH } from '../constants';
 
-export async function fetchAlertFields({
-  http,
-  featureIds,
-}: {
+export interface FetchAlertsFieldsParams {
+  // Dependencies
   http: HttpSetup;
+
+  // Params
+  /**
+   * Array of feature ids used for authorization and area-based filtering
+   */
   featureIds: ValidFeatureId[];
-}): Promise<FieldSpec[]> {
-  const { fields: alertFields = [] } = await http.get<{ fields: FieldSpec[] }>(
+}
+
+export const fetchAlertsFields = ({ http, featureIds }: FetchAlertsFieldsParams) =>
```

Comment:
should we add back the return type

Reply:
The generic type of the `get` method already properly types the return value, right? Do you prefer to specify it anyway?